### DISCUSSION TITLE
External events - conversion tracking

### DISCUSSION THREAD

**Discussion by Jaarson** [umami-software/umami#2866] [General]
Is there a way to capture external events?

For example, Lemon Squeezy sends me a webhook when a purchase is made. I can handle that webhook and send a custom event to Umami via API, but I think I might lose some data, such as the sessionID, right?

While the visitor is still on my website, I can pass any data to the Lemon Squeezy checkout, which then will be passed back in the webhook. However, can I obtain the user's sessionID somehow?

This is important for conversion tracking and I currently can't collect data about who (user or session) made a purchase.

Any ideas? 🙏

**Comment by Jaarson** [comment]
Idea: I could calculate sessionId manually on my end, using the code as below, and pass it to Lemon Squeezy or any other external place?

https://github.com/umami-software/umami/blob/88da20ea7f9e34a3d09cf8503ae09bff63b254bc/src/lib/session.ts#L70

**Comment by flowdee** [comment]
Same requirement here for our performance marketing tool.

<img width="2568" height="1492" alt="CleanShot 2025-12-04 at 10 38 36@2x" src="https://github.com/user-attachments/assets/d4e438d5-c14c-4d6a-922b-9c80ee7c11f9" />

**Comment by mikecao** [comment]
You should use distinct ids, https://umami.is/docs/distinct-ids
